### PR TITLE
feat: Team Management

### LINKED ISSUE #68: Team management
Obvious one! 

Manage your account and documents with your team <3

### PR DESCRIPTION
**Pull request by Aashish-Upadhyay-101**

### CODE REVIEW

**Comment by mfts** [comment]
Nice will check it put today 👀

File: prisma/schema.prisma
Comment (mfts) [inline prisma/schema.prisma:1]:
```
documents      Document[]
domains        Domain[]
plan           String     @default("free")
stripeId       String?    @unique // Stripe subscription / customer ID
subscriptionId String?    @unique // Stripe subscription ID
startsAt       DateTime? // Stripe subscription start date
endsAt         DateTime? // Stripe subscription end date
```

need to be part of the `Team` model. Essentially everything expect the login needs to be transferred to the `Team` model.

**Review by mfts — COMMENTED** [review]
Good foundation but more of the models need to be associated with `Team` account. 

There will be no individual user. Every user will have at least one team account. Everything is associated with the `Team` account and scoped by it. 

Think when logging into Vercel. 
Screenshot 2023-10-09 at 23 41 56
or unkey.dev 
Screenshot 2023-10-09 at 23 40 53


I'm happy to take a stab at it and build on your foundation.

File: prisma/schema.prisma
Comment (Aashish-Upadhyay-101) [reply prisma/schema.prisma:1]: @mfts got it, I will make changes and let you know

**Pull request closed without merging**